pandas/plotting/_core.py

```diff
@@ -1208,6 +1211,22 @@ def hist(self, by=None, bins=10, **kwargs):
             ...     columns = ['one'])
             >>> df['two'] = df['one'] + np.random.randint(1, 7, 6000)
             >>> ax = df.plot.hist(bins=12, alpha=0.5)
+
+        If `by` can be assigned by the DataFrame column names, or a list of column
+        names for which to group, and a grouped hist plot is generated:
+
+        .. plot::
+            :context: close-figs
+
+            >>> np.random.seed(159753)
+            >>> df = pd.DataFrame(np.random.randn(30, 2), columns=['A', 'B'])
+            >>> df['C'] = np.random.choice(['a', 'b', 'c'], 30)
+            >>> ax = df.plot.hist(column=['A', 'B'], by=['C'], figsize=(8, 10))
+
+        .. plot::
+            :context: close-figs
+
+            >>> ax = df.plot.hist(column=['A', 'B'], by='C', figsize=(8, 10))
```

Comment:
> Is providing `by='c'` returning different results than providing `by=['C']`? If this is returning the same, I would simply use `by='c'`. If they return different things, can you please explain (in the doc, not in the review).

thanks for your review, appreciate a lot! i think using capital `C` is because it is the column name in the example dataframe, while `c` is one of the values in `df['C']`, and we want to assign a column name to `by` argument.

> Do you think you can find an example with meaningful data, instead of just random numbers and letters? Also for the previous existing example (if it can be the same dataframe for both cases, much better). And if you don't mind adding a screenshot of this rendered in the PR, that would be very helpful to review how the plot looks like.

yeah, I will change, this is the example used in the original issue description, a clearer and more meaningful example is definitely better!!

Reply:
It was a typo in my comment. In both calls to `.plot.hist()` you're using uppercase `C`, the name of the column, but one inside a list, and one just the string. I think the output of both should be the same, and we shouldn't show the plot twice.

Reply:
ahh, okay, i was intended to show that it works both for column name, and a list of column names. But you are right, it is not very clear, so i removed it